dv/cosim/spike_cosim.cc

```diff
@@ -366,6 +366,19 @@ void SpikeCosim::set_mcycle(uint64_t mcycle) {
   processor->get_state()->mcycle->write(mcycle + 1);
 }
 
+void SpikeCosim::set_csr(const int csr_num, const uint32_t new_val) {
+  // If the default value, 0, did not change no need to write to the register.
+  // Note that this is tested with ibex-cosim-v0.3 version of Spike. 'set_csr'
+  // method might have a hardwired zero for mhpmcounterX registers.
+  if (new_val > 0) {
+#ifdef OLD_SPIKE
+    processor->set_csr(csr_num, new_val + 1);
+#else
+    processor->put_csr(csr_num, new_val + 1);
```

Comment:
Why the +1?

Reply:
It was because the performance counters in the spike side were the type `wide_counter_csr_t`. So, if we do a `unlogged_write` to them (which both `set_csr` and `put_csr` essentially do) they are getting decremented by one automatically. But you are right, handling it here does not make sense. I'll come up with a better way of doing it inside spike.